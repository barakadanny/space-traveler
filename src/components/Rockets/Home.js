import React from 'react';
import RockectsList from './RocketsList';

const Home = () => (
  <main>
    <RockectsList />
  </main>
);

export default Home;
