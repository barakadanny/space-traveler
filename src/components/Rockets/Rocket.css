.single-rocket {
  background-color: var(--white-color);
  display: flex;
  flex-direction: column;
  gap: 1rem;
  padding: 1rem;
  align-items: center;
}

.img-container {
  max-width: 200px;
  position: relative;
}

.img-container img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.rocket-info h3,
p {
  margin-bottom: 0.5rem;
}

/* Desktop view */
@media screen and (min-width: 768px) {
  .single-rocket {
    flex-direction: row;
  }
}
