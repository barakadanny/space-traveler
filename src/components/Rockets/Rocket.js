import React from "react";
import "./Rocket.css";

function Rocket() {
  return (
    <div>
      <div className="single-rocket">
        <div className="img-container">
          <img src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png" alt="" />
        </div>

        <div className="rocket-info">
          <h3>Falcon 1</h3>
          <p>
            First successful orbital launch vehicle in history, capable of
            delivering 13,450 kg to low Earth orbit. First successful orbital
            launch vehicle in history, capable of delivering 13,450 kg to low
            Earth orbit.
          </p>
          <button className="btn">Reserve Rocket</button>
        </div>
      </div>
    </div>
  );
}

export default Rocket;
