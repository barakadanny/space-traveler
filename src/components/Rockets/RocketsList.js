import React from 'react';
import Rocket from './Rocket';

function RockectsList() {
  return (
    <div className="rockets-list">
      <Rocket />
    </div>
  );
}

export default RockectsList;
