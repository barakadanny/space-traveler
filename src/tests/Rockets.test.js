import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Rockets from '../components/Rockets/RocketsList';

const mockStore = configureStore([]);

describe('Rockets', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rockets: [
        {
          id: 1,
          name: 'Falcon 1',
          description:
            'First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit. First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit.',
          image: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
        },
        {
          id: 2,
          name: 'Falcon 9',
          description:
            'First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit. First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit.',
          image: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
        },
      ],
    });
    component = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
