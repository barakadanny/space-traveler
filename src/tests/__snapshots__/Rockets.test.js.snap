// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets should render with given state from Redux store 1`] = `
<div
  className=""
>
  <div
    className="single-rocket"
  >
    <div
      className="img-container"
    >
      <img
        alt="rocket"
      />
    </div>
    <div>
      <h3>
        Falcon 1
      </h3>
      <p>
        First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit. First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit.
      </p>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Reserve rocket
      </button>
    </div>
  </div>
  <div
    className="single-rocket"
  >
    <div
      className="img-container"
    >
      <img
        alt="rocket"
      />
    </div>
    <div>
      <h3>
        Falcon 9
      </h3>
      <p>
        First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit. First successful orbital launch vehicle in history, capable of delivering 13,450 kg to low Earth orbit.
      </p>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Reserve rocket
      </button>
    </div>
  </div>
</div>
`;
